components/x-topic-search/src/TopicSearch.jsx

```diff
@@ -71,14 +72,23 @@ class TopicSearch extends Component {
 		}
 	}
 
-	handleInputClickOrFocus() {
+	checkSearchTermLength() {
```

Comment:
I would say this method is more about showing the result than checking the searchTerm length, but in any case, I think the following would be simpler, and avoids an additional class method:

```es6
	handleInputClick() {
		if (this.state.searchTerm.length >= this.minSearchLength) {
			this.setState({
				showResult: true
			});
		}
	}

	handleInputFocus(event) {
		event.target.select();
		this.handleInputClick();
	}
```